src/sentry/middleware/integrations/parsers/base.py

```diff
@@ -182,6 +185,42 @@ def get_response_from_webhookpayload_for_integration(
             regions=regions, identifier=integration.id, integration_id=integration.id
         )
 
+    def get_mailbox_identifier(self, integration: RpcIntegration, data: Mapping[str, Any]) -> str:
+        """
+        Used by integrations with higher hook volumes to create smaller mailboxes
+        that can be delivered in parallel. Requires the integration to implement
+        `mailbox_bucket_id`
+        """
+        # If we get fewer than 3000 in 1 hour we don't need to split into buckets
+        ratelimit_key = f"webhookpayload:{self.provider}:{integration.id}"
```

Comment:
Is this used by our stripe webhook handling in any way? Curious if we'll encounter an issue with that at all.

Reply:
It isn't. Parsers have to opt in to using `get_webhook_identifier` when creating outbox messages. So far stripe is always low volume and doesn't need additional partitioning.